tests/unit/test_bam_utils.py

```diff
@@ -106,3 +157,59 @@ def test_bam_header_missing_model(bam_header_without_program_group):
     assert ret == False
 
 
+@pytest.mark.parametrize("bam_header_with_multiple_program_groups", ['mas15teloprimev2'], indirect=True)
+def test_load_models_from_bam_header(bam_header_with_multiple_program_groups):
+    with tempfile.NamedTemporaryFile(delete=True) as f:
+        with pysam.AlignmentFile(f.name, "wb", header=bam_header_with_multiple_program_groups) as bf:
+            bf.close()
+
+        models = bam_utils.load_models(None, input_bam=f)
+
+        assert len(models) == 2
+        assert models[0].name == 'mas15v2'
```

Comment:
Can you do a test here for the actual json content of the models rather than just the name?  It'll keep us on track if we find that we have to change them.

Reply:
Fixed.